trace/setup.py

```diff
@@ -20,6 +20,7 @@
 install_requires = [
     'google-gax>=0.15.7, <0.16dev',
     'googleapis-common-protos[grpc]>=1.5.2, <2.0dev',
+    'wrapt',
```

Comment:
This is probably ok, but I want to make sure that we only add external dependencies rarely, when they are really needed.  Although it's been around for 3 years and has had several releases, I'm concerned that wrapt only has one maintainer.  It seems like we could use regular decorators and subclasses and get most of the way there.

Reply:
Removed the wrapt dependency, using subclasses works right now.